src/server/JasmineGraphInstanceService.cpp

```diff
@@ -1843,8 +1872,117 @@ void *instanceservicesession(void *dummyPt) {
                     write(connFd, chunk.c_str(), chunk.size());
                 }
             }
+        } else if (line.compare(JasmineGraphInstanceProtocol::INITIATE_FILES) == 0) {
+
+            instance_logger.log("Received from: " + JasmineGraphInstanceProtocol::INITIATE_FILES, "info");
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string trainData(data);
+
+            std::vector<std::string> trainargs = Utils::split(trainData, ' ');
+            instance_logger.log("trainData : " + trainData, "info");
+
+            string graphID;
+            string partitionID = trainargs[trainargs.size() - 1];
+
+            for (int i = 0; i < trainargs.size(); i++) {
+                if (trainargs[i] == "--graph_id") {
+                    graphID = trainargs[i + 1];
+                    break;
+                }
+            }
+            instance_logger.log("graphID : " + graphID, "info");
+
+            std::thread *workerThreads = new std::thread[2];
+            workerThreads[0] = std::thread(&JasmineGraphInstanceService::createPartitionFiles, graphID, partitionID,
+                                           "local");
+            workerThreads[1] = std::thread(&JasmineGraphInstanceService::createPartitionFiles, graphID, partitionID,
+                                           "centralstore");
+
+            for (int threadCount = 0; threadCount < 2; threadCount++) {
```

Comment:
@kasundharmadasa  Why do we have only two threads here?

Reply:
This is for creating partition files for localstore and central store.

Reply:
Shall we remove the for loop then? Just have two lines like follows,

```
 workerThreads[0].join();

 workerThreads[1].join();
```

Reply:
Updated